route/router.go

```diff
@@ -39,6 +40,41 @@ type Router interface {
 	Plan() store.Variable[Plan]
 }
 
+// PartialPlan is an (incomplete) Plan that operates on the internal
+// solver data structures. Certain router options that customize solver
+// internals have to work with this data structure.
+// It gives access to the following information:
+//  - Unplanned(): an Integer Domain with not yet assigned or unassigned
+//	  stops indexes.
+//  - Unassigned(): an Integer Domain with unassigned stop indexes.
+//	  These are stops explicitly excluded from being served by a vehicle.
+//  - Vehicles(): a slice of vehicles part of this partial plan.
+//  - Value(): the value of this plan.
+type PartialPlan interface {
+	Unplanned() model.Domain
```

Comment:
We tend to organize the interface methods in alphabetical order. We try, at least.